content/go/concepts/interfaces/interfaces.md

```diff
@@ -0,0 +1,268 @@
+---
+Title: 'Interfaces'
+Description: 'An interface is a set of method signatures.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Control Flow'
+  - 'Interfaces'
+CatalogContent:
+  - 'learn-go'
+  - 'paths/back-end-engineer-career-path'
+---
+
+An interface is composed of set of method signatures. These method signatures define the input and return values of which a data type or struct can conform to. In order to implement an interface, the type or struct must satisfy all the defined method signatures.
+
+## Creating an Interface
+
+```go
+type InterfaceName interface {
+    method() return_type
+}
+```
+
+A type implements an interface by implementing its methods, In `Go`, interfaces are implemented implicitly and enables polymorphism.
```

Comment:
@Pavan-04-08 One more thing. Can we first go over an example of a simple implementation and move this part about polymorphism below? In the next iteration we can go into polymorphism a bit more